### PR TITLE
[FIXED JENKINS-47071] Replace BooleanClosureWrapper with serializable

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-47071](https://issues.jenkins-ci.org/browse/JENKINS-47071)

BooleanClosureWrapper isn't serializable, so attempts to serialize it
(i.e., saveProgram) result in errors. This replaces
BooleanClosureWrapper with CpsBooleanClosureWrapper, which *is*
serializable.

No point in testing here, but actual test coming downstream.

cc @reviewbybees

### CODE REVIEW

File: lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsBooleanClosureWrapper.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * Bridge for a call based on a map entry. If the call is done on a {@link Closure}
+     * taking one argument, then we give in the {@link Map.Entry}, otherwise we will
+     * give in the key and value.
+     */
+    public <K,V> boolean callForMap(Map.Entry<K, V> entry) {
+        if (numberOfArguments==2) {
+            return call(entry.getKey(), entry.getValue());
+        } else {
+            return call(entry);
```
Comment (jglick) [inline lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsBooleanClosureWrapper.java]: Maybe just create a new `BooleanClosureWraper` and delegate to it, so we have less code here?

Comment (abayer) [reply lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsBooleanClosureWrapper.java]: Good point.

**Comment by abayer** [comment]
@jglick I have pretty much no idea what you mean in re `PickleFactory`. =)

**Comment by svanoort** [comment]
I have honestly no idea about how this works, so I probably shouldn't be reviewing it :-/

**Pull request merged**